src/cli.rs

```diff
@@ -3,20 +3,30 @@ use reqwest::blocking as reqwest;
 use serde_json;
 use std::fs;
 use std::io::{self, Read, Write};
+use std::net::TcpListener;
+use std::net::TcpStream;
 use std::os::unix;
 use threadpool::ThreadPool;
 
-// TODO: things to configure:
-//  - infura project id (not just env var?)
-//  - rpc endpoint port
-//  - rpc endpoint type (tcp, ws, ipc)
-//  - max concurrent requests (ie~ threadpool size)
+#[derive(Debug, PartialEq)]
+pub struct TrinConfig {
+    pub protocol: String,
+    pub infura_project_id: String,
+    pub endpoint: u32,
+    pub pool_size: u32,
+}
 
-pub fn launch_trin(infura_project_id: String) {
-    println!("Launching with infura key: '{}'", infura_project_id);
+pub fn launch_trin(trin_config: TrinConfig) {
+    let pool = ThreadPool::new(trin_config.pool_size as usize);
 
-    let pool = ThreadPool::new(2);
+    match &trin_config.protocol[..] {
+        "ipc" => launch_ipc_client(pool, trin_config),
```

Comment:
Maybe pass in `trin_config.infura_project_id` instead, so the downstream methods don't need to depend on `TrinConfig`.

Reply:
👍 made the change for `launch_ipc_client` but left the args for `launch_http_client` as is since that function also needs access to `trin_config.endpoint`.